src/Internal/DependencyManagement/ExtendedContainer.php

```diff
@@ -0,0 +1,102 @@
+<?php
```

Comment:
We should rename this file as `Container` and existing `Container` to `ContainerProvider` to prevent any confusion between which container is being used (and also based on what these classes are actually doing).

Reply:
I think since our `Container` class conforms to the PSR-11 `Container` interface it makes sense to leave the name as-is there.

Reply:
👍  and also the only thing devs need to worry about is that they will interact with an instance of `ContainerInterface` (to get objects) or with a service provider (to register classes), so I don't think it will be confussing.